app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx

```diff
@@ -0,0 +1,409 @@
+    };
+    if (window.wrapEnd) {
+      timelineWindow.displayEnd = new Date(2000, 1, 1, 23, 59, 59, 999);
+    }
+    return timelineWindow;
+  }
+
+  function getWindowStyle(window: IWindow): any {
+    const dayMinutes = 24 * 60;
+    const start = window.startHour * 60 + window.startMin,
+      end = window.endHour * 60 + window.endMin,
+      width = ((end - start) / dayMinutes) * 100,
+      startOffset = (start / dayMinutes) * 100;
+
+    return {
+      width: width + '%',
+      left: startOffset + '%',
+    };
+  }
+
+  return (
+    <div className="execution-windows">
+      <div className="form-group">
+        <div className="col-md-9 col-md-offset-1">
+          {isWindowExpression && (
+            <div className="row" style={{ marginTop: '20px' }}>
+              <strong> Restrict execution to specific time windows when this expression is true</strong>
+              <input type="text" className="form-control input-sm" />
+              <TextInput
+                value={props.restrictExecutionDuringTimeWindow}
+                onChange={(e: React.ChangeEvent<any>) =>
+                  props.updateStageField({ restrictExecutionDuringTimeWindow: e.target.value })
+                }
+              />
+            </div>
+          )}
+          {!isWindowExpression && (
+            <div className="checkbox">
+              <CheckboxInput
+                text={<strong> Restrict execution to specific time windows</strong>}
+                value={props.restrictExecutionDuringTimeWindow}
+                onChange={() =>
+                  props.updateStageField({
+                    restrictExecutionDuringTimeWindow: !props.restrictExecutionDuringTimeWindow,
+                  })
+                }
+              />
+            </div>
+          )}
+        </div>
+      </div>
+      {props.restrictExecutionDuringTimeWindow && (
+        <div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <h5>
+                <strong>Days of the Week</strong>
+                <small>(No days selected implies execution on any day if triggered)</small>
+              </h5>
+              <div className="execution-window-days-of-week">
+                <ExecutionWindowDayPicker days={days} onChange={windowDaysUpdated} />
+              </div>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <p>
+                <strong>Time of Day</strong>
+              </p>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <div className="execution-window-graph">
+                <div className="execution-day">
+                  {timelineWindows.map((w, i) => {
+                    return (
+                      <Tooltip key={i} value={moment(w.start).format('HH:mm') + ' - ' + moment(w.end).format('HH:mm')}>
```

Comment:
We moved off momentjs a few months ago (I think it's still being brought in by some transient dependencies, which is why it's available here). Can you switch this to either `date-fns` or `luxon`? Both have a decent `format` method that should be basically compatible.

Reply:
👍